module/planning/PlanKick/src/PlanKick.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef MODULE_PLANNING_PLANKICK_HPP
+#define MODULE_PLANNING_PLANKICK_HPP
+
+#include <extension/Behaviour.hpp>
+#include <nuclear>
+#include <string>
+
+namespace module::planning {
+
+    class PlanKick : public ::extension::behaviour::BehaviourReactor {
+    private:
+        /// @brief Stores configuration values
+        struct Config {
+            float timeout_threshold       = 0.0;
+            float ball_distance_threshold = 0.0;
+            float ball_angle_threshold    = 0.0;
+            bool align                    = false;
+            float target_angle_threshold  = 0.0;
+            std::string kick_leg          = "";
```

Comment:
Make this an enum (or smart enum I think we have a leg side enum?)

Reply:
Does yaml support enums?